datasets/ethos/ethos.py

```diff
@@ -0,0 +1,159 @@
+            version=datasets.Version("1.0.0", ""),
+            description="Ethos Binary",
+            variation="binary",
+        ),
+        EthosConfig(
+            name="multilabel",
+            version=datasets.Version("1.0.0", ""),
+            description="Ethos Multi Label",
+            variation="multilabel",
+        ),
+    ]
+
+    def _info(self):
+        if self.config.variation == "binary":
+            f = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "label": datasets.features.ClassLabel(names=["no_hate_speech", "hate_speech"]),
+                }
+            )
+        else:
+            f = datasets.Features(
+                {
+                    "text": datasets.Value("string"),
+                    "violence": datasets.ClassLabel(names=["not_violent", "violent"]),
+                    "directed_vs_generalized": datasets.ClassLabel(names=["generalied", "directed"]),
+                    "gender": datasets.ClassLabel(names=["false", "true"]),
+                    "race": datasets.ClassLabel(names=["false", "true"]),
+                    "national_origin": datasets.ClassLabel(names=["false", "true"]),
+                    "disability": datasets.ClassLabel(names=["false", "true"]),
+                    "religion": datasets.ClassLabel(names=["false", "true"]),
+                    "sexual_orientation": datasets.ClassLabel(names=["false", "true"]),
+                }
+            )
+        return datasets.DatasetInfo(
+            features=f,
+            supervised_keys=None,
+            homepage="https://github.com/intelligence-csd-auth-gr/Ethos-Hate-Speech-Dataset/tree/masterethos/ethos_data",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        if self.config.variation == "binary":
+            url = {
+                "train": "https://raw.githubusercontent.com/intelligence-csd-auth-gr/Ethos-Hate-Speech-Dataset"
+                "/master/ethos/ethos_data/Ethos_Dataset_Binary.csv"
+            }
+        else:
+            url = {
+                "train": "https://raw.githubusercontent.com/intelligence-csd-auth-gr/Ethos-Hate-Speech-Dataset"
+                "/master/ethos/ethos_data/Ethos_Dataset_Multi_Label.csv"
+            }
+        downloaded_files = dl_manager.download_and_extract(url)
+
+        return [datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": downloaded_files["train"]})]
+
+    def _generate_examples(self, filepath):
+        """Yields examples."""
+
+        data = pd.read_csv(filepath, delimiter=";")
+        if self.config.variation == "binary":
+
+            x = data["comment"].values
+            y = [1 if i >= 0.5 else 0 for i in data["isHate"].values]
+            class_names = ["no_hate_speech", "hate_speech"]
+            for i in range(len(x)):
+                _id = i
+                yield _id, {"text": x[i], "label": class_names[y[i]]}
```

Comment:
According to this code though, the `hate_speech` label correspond to 1.
Is this expected ?

Reply:
Yes! I meshed up again the paragraph in ReadMe 0 is without hate speech and 1 is with hate speech. I will update the readme immediately. Sorry

Reply:
``` with a `text` containing the  text and a `label` describing if a comment contains hate speech content (1 - hate-speech) or not (0 - non-hate-speech).  ``` Fixed it!

Reply:
And in case this example is tricky: 'What the fuck stupid people !!!', it is because it is abusive, and offensive, etc, but it does not contain hate speech content (according to the definition of hate speech).

Reply:
Ok thanks !
Also in the dataset card the current example shows `no_hate_speech` while the text seems to actually show hate speech. Could we have an example where the label is correct instead ?

Reply:
Oh I see !

Reply:
Then maybe you can add a note below the example to explain this if you don't mind ?
I was a bit confused by the label ^^

Reply:
Okay! I'm on it! I will add 1 more example as well!

Reply:
@lhoestq is it okay now?

Reply:
Thanks ! It's clearer now :)